code/game/machinery/computer/security.dm

```diff
@@ -1,46 +1,321 @@
+				return
+
+			set_active_record_security(CreateSecurityRecord(active_record_general.fields["name"], active_record_general.fields["id"]))
+
+		if ("delete_security_record")
+			if (!istype(active_record_security, /datum/data/record) || !sanity_check(params["ref"]))
+				return
+
+			GLOB.data_core.security -= active_record_security
+			qdel(active_record_security)
+
+			var/mob_gid = hex2num(active_record_general.fields["id"])
+			for(var/mob/living/carbon/human/human as anything in GLOB.human_mob_list)
+				if (human.gid == mob_gid)
+					human.sec_hud_set_security_status()
+					break
+
+		if ("delete_all_security_records")
+			for(var/datum/data/record/record as anything in GLOB.data_core.security)
+				GLOB.data_core.security -= record
+				qdel(record)
+
+			for(var/mob/living/carbon/human/human as anything in GLOB.human_mob_list)
+				human.sec_hud_set_security_status()
+
+		if ("set_active_record")
+			var/datum/data/record/general_record = locate(params["ref"])
+			if (!istype(general_record))
+				return
+
+			set_active_record_general(general_record)
+			set_active_record_security(null)
+
+			// Get active security record, if one exists.
+			for(var/datum/data/record/security_record as anything in GLOB.data_core.security)
+				if (security_record.fields["id"] == general_record.fields["id"])
+					set_active_record_security(security_record)
+					break
+
+		if ("edit_name")
+			if (!sanity_check(params["ref"]))
+				return
+
+			var/sanitised_name = reject_bad_name(tgui_input_text(ui.user, "Enter a new name", "New Name", active_record_general.fields["name"]))
+			if (!sanitised_name || !sanity_check(params["ref"]))
+				return
+
+			message_admins("[key_name(ui.user)] has changed the record name of [active_record_general.fields["name"]] to [sanitised_name]")
+			active_record_general.fields["name"] = sanitised_name
+
+		if("toggle_sex")
+			if (!sanity_check(params["ref"]))
+				return
+
+			if (active_record_general.fields["sex"] == "male")
+				active_record_general.fields["sex"] = "female"
+			else
+				active_record_general.fields["sex"] = "male"
+
+		if("edit_age")
+			if (!sanity_check(params["ref"]))
+				return
+
+			// New records have a string in their age field which will break this input.
+			var/default_age = active_record_general.fields["age"] == "Unknown" ? 0 : active_record_general.fields["age"]
+
+			var/input = tgui_input_number(ui.user, "Enter a new age", "New Age", default_age, AGE_MAX, AGE_MIN)
+			if (!input || !sanity_check(params["ref"]))
+				return
+
+			active_record_general.fields["age"] = input
+
+		if ("set_criminal_status")
+			if (!istype(active_record_security, /datum/data/record) || !sanity_check(params["ref"]))
+				return
+
+			var/new_wanted_status = params["new_status"]
+			if(!new_wanted_status || !(new_wanted_status in WANTED_STATUSES))
+				return
+
+			active_record_security.fields["criminal"] = new_wanted_status
+
+			var/mob_gid = hex2num(active_record_general.fields["id"])
```

Comment:
I'm wondering, do we want to message admins, as a heads up, that someone has been flagged for arrest? It could help get eyes on a situation before it escalates too far